File: Assets/Scripts/Spells/Priest/GuidingBolt.cs
using UnityEngine;

public class GuidingBolt : Spell {

    public override void Cast(Character caster, Character target) {
        if (target.currentHealth > 0) {
            CombatManager.instance.Damage(target, damage);
            caster.currentMana -= manaCost;
        }
    }

    public override void UpdateDescription() {
        description = $"The target takes {damage} damage.";
    }
}

File: Assets/Scripts/Lua/Spells/Priest/GuidingBolt.lua
function Cast(caster, target)
    if target == nil or target.currentHealth > 0 then
        caster.currentMana = caster.currentMana - spell.manaCost
        combatManager.Damage(target, spell.damage + caster.intelligence, caster)
        caster.Sync()
    else
        log("Invalid Target.")
    end
end

function UpdateDescription()
    description = "The target takes", spell.damage, "damage.";
end